x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/IndexLifecycleOriginationDateParserTests.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ilm;
+
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.test.ESTestCase;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Locale;
+import java.util.TimeZone;
+
+import static org.elasticsearch.xpack.core.ilm.IndexLifecycleOriginationDateParser.parseIndexNameAndExtractDate;
+import static org.elasticsearch.xpack.core.ilm.IndexLifecycleOriginationDateParser.shouldParseIndexName;
+import static org.hamcrest.Matchers.is;
+
+public class IndexLifecycleOriginationDateParserTests extends ESTestCase {
+
+    public void testShouldParseIndexNameReturnsFalseWhenOriginationDateIsSet() {
+        Settings settings = Settings.builder()
+            .put(LifecycleSettings.LIFECYCLE_ORIGINATION_DATE, 1L)
+            .build();
+        assertThat(shouldParseIndexName(settings), is(false));
+    }
+
+    public void testShouldParseIndexNameReturnsFalseIfParseOriginationDateIsDisabled() {
+        Settings settings = Settings.builder()
+            .put(LifecycleSettings.LIFECYCLE_PARSE_ORIGINATION_DATE, false)
+            .build();
+        assertThat(shouldParseIndexName(settings), is(false));
+    }
+
+    public void testShouldParseIndexNameReturnsTrueIfParseOriginationDateIsTrueAndOriginationDateIsNotSet() {
+        Settings settings = Settings.builder()
+            .put(LifecycleSettings.LIFECYCLE_PARSE_ORIGINATION_DATE, true)
+            .build();
+        assertThat(shouldParseIndexName(settings), is(true));
+    }
+
+    public void testParseIndexNameThatMatchesExpectedFormat() throws ParseException {
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy.MM.dd", Locale.getDefault());
+        dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+        long expectedDate = dateFormat.parse("2019.09.04").getTime();
+
+        {
+            long parsedDate = parseIndexNameAndExtractDate("indexName-2019.09.04");
+            assertThat("indexName-yyyy.MM.dd is a valid index format", parsedDate, is(expectedDate));
+        }
+
+        {
+            long parsedDate = parseIndexNameAndExtractDate("indexName-2019.09.04-0000001");
```

Comment:
Can you add a test for `index-2019.09.04-2018.01.01-002` (or something like it) to make sure it uses the correct date in the name?

Reply:
Adding a test for this but this will currently fail by picking `2019.09.04-2018.01.01` as the date which can't be parsed. Do you think we should do a more extensive validation here? (eg. max 2 dashes in the name)

Adding a test for `indexName-2019.09.04-2019.09.24` to make sure we fetch the correct date (`2019.09.04` in this case)

Reply:
> Adding a test for this but this will currently fail by picking `2019.09.04-2018.01.01` as the date which can't be parsed. Do you think we should do a more extensive validation here? (eg. max 2 dashes in the name)

This seems like a bug then, a user should be free to have any valid index name prior to `-yyyy.MM.dd(-[\\d]+)?`, with as many dashes as they'd like. It seems like for an index named `index-2019.09.04-2018.01.01-002` it should extract `2018.01.01` as the origination date, right?

Reply:
I think you're right, pushing a fix shortly